wasm/src/main/scala/org/scalajs/linker/backend/webassembly/Instructions.scala

```diff
@@ -98,6 +98,30 @@ object Instructions {
       val fieldIdx: FieldIdx
   ) extends Instr(mnemonic, opcode)
 
+  sealed abstract class MemoryInstr(
+      mnemonic: String,
+      opcode: Int,
+      val memoryName: MemoryName
+  ) extends Instr(mnemonic, opcode)
+
+  case class MemoryArg private (offset: Int, align: Int)
+  object MemoryArg {
+
+    /** We use 0 for `memarg.align` (an expected alignment) for now. Which means we don't provide no
+      * hint to a WebAssembly implementation regarding the memory alignment, and memory access may
+      * be slower on some hardware. However, unaligned access is still allowed in WebAssembly.
+      *
+      * @see
+      *   https://www.w3.org/TR/wasm-core-2/#memory-instructions%E2%91%A4
+      */
+    def apply(): MemoryArg = MemoryArg(0, 0)
```

Comment:
Memory instructions receive it's arguments both from operand and it's immediate arguments (`MemoryArg`).

For example, the following wasm code will load `i32` value from memory offset `300( = 100 + 200)`.

```wasm
i32.const 100
i32.load offset=200 align=0
```

Since we set `offset` of the immediate memory arg to be  `0`, the offset will be gall given from the operand.